generic3g/specs/FrequencyAspect.F90

```diff
@@ -0,0 +1,140 @@
+#include "MAPL_Generic.h"
+module mapl3g_FrequencyAspect
+   use mapl3g_StateItemAspect
+   use mapl3g_AccumulatorActionInterface
+   use esmf
+   implicit none
+   private
+
+   public :: FrequencyAspect
+
+   type, extends(StateItemAspect) :: FrequencyAspect
+      type(ESMF_TimeInterval) :: dt
+      character(len=:), allocatable :: accumulation_type
+   contains
+      procedure :: matches
+      procedure :: supports_conversion_general
+      procedure :: supports_conversion_specific
+      procedure :: make_action
+   end type FrequencyAspect
+
+   interface FrequencyAspect
+      module procedure :: construct_frequency_aspect
+   end interface FrequencyAspect
+
+   interface operator(.divides.)
+      module procedure :: divides
+   end interface operator(.divides.)
+
+contains
+
+   function construct_frequency_aspect(dt, accumulation_type) result(aspect)
+      type(FrequencyAspect) :: aspect
+      type(ESMF_TimeInterval), optional, intent(in) :: dt
+      character(len=*), optional, intent(in) :: accumulation_type
+
+      associate(adt => aspect%dt, atype => aspect%accumulation_type)
+         atype = NO_ACCUMULATION
```

Comment:
I wonder if "INSTANTANEOUS" might be clearer than "NO_ACCUMULATION"?

Reply:
"NO_ACCUMULATION" means that no accumulation has been set. "INSTANTANEOUS" means that the timesteps are the same. (I know that instantaneous usually implies 0 duration.) You could have accumulation set (not "NO_ACCUMULATION") and equal timesteps. That is not prohibited currently, but it doesn't make a lot of sense. `accumulation_type` is set for the destination `FieldSpec`. It is irrelevant for the source `FieldSpec` as it relates to the coupling between them. I'm not wedded to the names, but `accumulation_type` is set in the constructor for `FieldSpec`, so I used "NO_ACCUMULATION" when `accumulation_type` is not passed in the constructor, since it describes the `FieldSpec` by itself.

Reply:
changed